<h2>Ejercicio 3<h2>

In [1]:
import pandas as pd
from wget import download
from os import path, listdir

<h3>Inciso a

In [2]:
#descargamos si no existen
if not path.exists("catalogo_streaming_argentina_2024.xlsx"):
    download("https://ignaciorlando.github.io/datasets/data-science/catalogo_streaming_argentina_2024.xlsx")
else:
    print("No vamos a bajar el archivo porque ya existe")

In [ ]:
df = pd.read_excel('catalogo_streaming_argentina_2024.xlsx')
print(df)

In [27]:
#cabecera
print(df.info())

<class 'pandas.DataFrame'>
RangeIndex: 19817 entries, 0 to 19816
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           19817 non-null  string 
 1   platform        19817 non-null  str    
 2   type            19817 non-null  str    
 3   imdbVotes       17854 non-null  float64
 4   imdbScore       17927 non-null  float64
 5   tmdbPopularity  19538 non-null  float64
 6   tmdbScore       18954 non-null  float64
 7   runtime         19817 non-null  int64  
 8   genres          19665 non-null  str    
 9   director        14626 non-null  str    
 10  year_range      19817 non-null  object 
 11  url             19817 non-null  str    
dtypes: float64(4), int64(1), object(1), str(5), string(1)
memory usage: 1.8+ MB
None


In [26]:
#primeros 5 registros
print(df.tail(5))

            title            platform   type  imdbVotes  imdbScore  \
19812    Три кота             Netflix   SHOW        NaN        NaN   
19813  أم الدراهم  Amazon Prime Video   SHOW        NaN        NaN   
19814   ورق مسموم  Amazon Prime Video   SHOW        NaN        NaN   
19815        ガクドリ  Amazon Prime Video  MOVIE        NaN        NaN   
19816  新歌舞伎：赤胴鈴之助             Netflix  MOVIE        NaN        NaN   

       tmdbPopularity  tmdbScore  runtime           genres       director  \
19812          13.684        5.4        5  Ciencia ficción            NaN   
19813           0.858        NaN       29          Comedia            NaN   
19814           0.600        NaN       48              NaN            NaN   
19815           2.841        NaN       90              NaN        Kei Era   
19816           3.203        NaN      169         Fantasía  Kikunojo Onoe   

      year_range                                                url  
19812  2013_2017              http://www.netfl

In [11]:
df.describe(include='all')

,title,platform,type,imdbVotes,imdbScore,tmdbPopularity,tmdbScore,runtime,genres,director,year_range,url
count,19817,19817,19817,1.785400e+04,17927.000000,19538.000000,18954.000000,19817.000000,19665,14626,19817.0,19817
unique,18008,6,2,NaN,NaN,NaN,NaN,NaN,4189,8812,27.0,19062
top,Perdida,Netflix,MOVIE,NaN,NaN,NaN,NaN,NaN,Documental,Steven Spielberg,2022.0,http://www.netflix.com/title/81665878
freq,5,7109,14060,NaN,NaN,NaN,NaN,NaN,1277,27,1697.0,2
mean,NaN,NaN,NaN,4.838858e+04,6.399520,41.195589,6.461363,80.542867,NaN,NaN,NaN,NaN
std,NaN,NaN,NaN,1.488473e+05,1.200258,131.031366,1.254487,37.092943,NaN,NaN,NaN,NaN
min,NaN,NaN,NaN,5.000000e+00,1.100000,0.001000,0.500000,0.000000,NaN,NaN,NaN,NaN
25%,NaN,NaN,NaN,4.620000e+02,5.700000,5.433250,5.889500,48.000000,NaN,NaN,NaN,NaN
50%,NaN,NaN,NaN,2.849500e+03,6.500000,14.428500,6.600000,89.000000,NaN,NaN,NaN,NaN
75%,NaN,NaN,NaN,2.196975e+04,7.300000,33.312500,7.273750,105.000000,NaN,NaN,NaN,NaN


<h3>Inciso b

<p>Analisis<p>

- title: Representa el titulo del contenido - Cualitativo nominal - El titulo mas mencionado es 'Perida' con frecuencia 5.

- platform: Representa la plataforma donde se encuentra el contenido - Cualitativo nominal - La plataforma mas mencionada es 'Netflix' con frecuencia 7109.
- type: Representa la clasificacion del contenido - Cualitativo nominal - El tipo mas mencionado es 'MOVIE' con frecuencia 14060. 
- imdbVotes: Representa la cantidad de votantes que calificaron el contenido - Cuantitativa discreta.
- imdbScore: Representa el promedio ponderado de clasificaciones dadas por usuarios al contenido - Cuantitativa continua (1 a 10). 
- tmdbPopularity: Representa el nivel de popularidad por tmdb - Cuantitativa continua.
- tmdbScore: Representa la clasificacion promedio del contenido - Cuantitativo continuo (1 a 10).  
- runtime: Representa el tiempo de duracion en minutos - Cuantitativa discreta.
- genres: Representa el genero del contenido - Cualitativo nominal - El genero mas mencionado es 'Documental' con frecuencia 1277.
- director: Representa el director del contenido - Cualitativo nominal - El director mas mencionado es 'Steven Spielberg' con frecuencia 27.
- year_range: Indica el rango de anios activo del contenido - Cualitativo ordinal - El anio con mayores contenidos activo fue 2022 con frecuencia 1697.
- url: Represeta la URL del contenido.

In [ ]:
dfPerdida = df[df['title']=='Perdida']
print(dfPerdida)

<h3>Inciso c

In [ ]:
#cambiamos el title a string.
import string
# 1. Asignar únicamente a la columna deseada
df['title'] = df['title'].astype('string')
#print(df["year_range"].apply(type).value_counts())

<p>1. Cuales son las peliculas con mejor proporcion imdbScore con imdbVotes

<p>Para eso usamos

- Escala logaritmica: Damos validez a como la cantidad de votantes se incrementa de a numero de ceros.

- Reescalado a [0, 1]: $$ \left( \frac{\log_{10}(v) - \log_{10}(v_{min})}{\log_{10}(v_{max}) - \log_{10}(v_{min})} \right) $$

- Reescalado a [0, 10]: $$ \textbf{votos\_score} = 10 \cdot \left( \frac{\log_{10}(v) - \log_{10}(v_{min})}{\log_{10}(v_{max}) - \log_{10}(v_{min})} \right) $$

- Ponderamos: 
    $$ (0.7 \cdot \text{imdbScore}) + (0.3 \cdot \text{votes\_scaled\_1\_10}) $$

In [36]:
import numpy as np
log_votos = np.log10(df['imdbVotes']+1)
min_log = log_votos.min()
max_log = log_votos.max()
df['votes_scaled'] = 10 * ((log_votos - min_log) / (max_log - min_log))
df['rnk_rating'] = (0.7 * df['imdbScore']) + (0.3 * df['votes_scaled'])

df.sort_values(by='rnk_rating', ascending=False).head(3)


,title,platform,type,imdbVotes,imdbScore,tmdbPopularity,tmdbScore,runtime,genres,director,year_range,url,votes_scaled,rnk_rating
7,Breaking Bad,Netflix,SHOW,2166167.0,9.5,3024.965,8.908,47,"Drama, Crimen, Misterio & Suspense",NaN,2006_2008,http://www.netflix.com/title/70143836,9.775484,9.582645
0,Cadena perpetua,Max,MOVIE,2906253.0,9.3,159.594,8.705,142,"Drama, Crimen",Frank Darabont,1900_2002,https://play.max.com/movie/9b4dacba-2f80-4272-...,10.000000,9.510000
5,Juego de tronos,Max,SHOW,2309345.0,9.2,1391.340,8.449,58,"Fantasía, Acción & Aventura, Drama, Ciencia fi...",NaN,2011_2015,https://play.max.com/show/4f6b4985-2dc9-4ab6-a...,9.824378,9.387313


2. Cuales son las plataformas, que en promedio, tienen la mejor rnk_rating?

In [42]:
grouped_data = df.groupby('platform')
media_platform = grouped_data['rnk_rating'].median()
print(media_platform.sort_values(ascending=False))
#media_platfrom es una serie (key, valor): (platform, media_rnk_rating)

platform
Apple TV Plus         6.482214
Max                   6.254675
Disney Plus           6.252962
Netflix               6.030112
Paramount Plus        5.967747
Amazon Prime Video    5.522202
Name: rnk_rating, dtype: float64
